.golangci.yml

```diff
@@ -1,193 +1,121 @@
-run:
```

Comment:
Using `golangci-lint` tool to migrate the config file from v1 to v2. Some fields are not supported, but after I ran the `make reviewable` there are no issues and pass the lint step.

Reply:
> Due to build submodules, in order to use the new version of golangci-lint for reviewable do:
> GOLANGCILINT_VERSION=2.1.2 make reviewable

let's go ahead and set the golangci version we want to use now in the [MAKEFILE](https://github.com/crossplane/provider-template/blob/main/Makefile#L14-L22), e.g.:
```
GOLANGCILINT_VERSION = 2.1.2
```

Reply:
Nice! Will add it!